jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/BaseGenerator.java

```diff
@@ -289,4 +294,19 @@ protected void addSchemaLabels(BuildConfiguration.BuildConfigurationBuilder buil
             buildBuilder.labels(labels);
         }
     }
-}
+
+    public void checkMissingArtifacts() {
+        try {
+            final File finalOutputArtifact = JKubeProjectUtil.getFinalOutputArtifact(getProject());
+            if(finalOutputArtifact == null) {
+                log.error("Final output artifact file was not detected. The project may have not been built. HINT: try to compile and package your application prior to running the container image build task.");
+            } else if (isArtifactSameAsPrevious(finalOutputArtifact, finalOutputArtifact.getParentFile().toPath(), log)) {
+                log.warn( "Final output artifact is the same as previous build. You might have forgotten to compile and package your application after making changes.");
```

Comment:
you should have the main logic here not in the utils method, Maybe:
1. calculateCurrentOutputArtifactChecksum
2. getPreviousOutputArtifactChecksum
3. saveCurrentOutputArtifactChecksum
4. compare